### PR TITLE
Upgrade go to 1.21

### PR DESCRIPTION
**Pull request by inteon**
Upgrade all go versions & go.mod files to 1.21

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/cert-manager/trust-manager
 
-go 1.20
+go 1.21.1
```
Comment (SgtCoDFish) [inline go.mod:3]: suggestion: this feels wrong? If we're giving a specific patch release it should surely be 1.21.3 (latest) - but I'd think we should just use `1.21` instead, right?

File: hack/tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/cert-manager/tools/hack/tools
 
-go 1.20
+go 1.21.1
```
Comment (SgtCoDFish) [inline hack/tools/go.mod:3]: suggestion: as above!

File: trust-packages/debian/go.mod [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module github.com/cert-manager/trust-manager/debian-bundle-static
 
-go 1.19
+go 1.21.1
```
Comment (SgtCoDFish) [inline trust-packages/debian/go.mod:3]: suggestion: as above!

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (erikgb) [inline go.mod:3]: Any particular reason for selecting 1.21.1? Why not 1.21.0 (first 1.21.x) or 1.21.3 (current latest).

Comment (erikgb) [reply go.mod:3]: I think you cannot set just 1.21 anymore.

Comment (inteon) [reply go.mod]: Ah, sorry; this was quite an old change and 1.21.1 was the latest version at that point in time. I updated the PR to 1.21.3.

Comment (inteon) [reply go.mod]: I updated it to 1.21.3

**Review by erikgb — APPROVED** [review]
/lgtm

But we might consider requiring just 1.21.0 in the go.mod files. Do we plan to bump the requirement as new Go releases are released?

**Comment by inteon** [comment]
I don't think we have to bump it every time, I just did it now because in 1.21 the go.sum files contain more entries (I think they were missing in 1.20)

**Comment by SgtCoDFish** [comment]
Closing the loop: https://go.dev/ref/mod#go-mod-file-go

The go directive is specifying a minimum. The fact that it's an older version here isn't a problem really!

**Pull request merged**